pkg/datastore/datastore.go

```diff
@@ -28,15 +28,25 @@ func New() *memoryDatastore {
 
 // Reader is the interface that wraps the Read method.
 type Reader interface {
-	// Read takes a package identifier and returns metadata defined in the opsrc
+	// Read takes an OperatorSource name and package identifier and returns metadata defined in the opsrc
 	// spec to associate that package to a specific opsrc in the marketplace.
 	// The OpsrcRef returned can be used to determine how to download manifests
 	// for a specific operator package.
-	Read(packageID string) (opsrcMeta *OpsrcRef, err error)
+	Read(source, packageID string) (opsrcMeta *OpsrcRef, err error)
```

Comment:
Do we really need two sets of APIs for this? Moreover if you are doing this for API compatibility then you should leave the old API as is and introduce a new API for the new functionality.

Reply:
I spoke with @kevinrizza earlier today and he made the argument that the datastore shouldn't perform logic to decide how to retrieve data and that it is the responsibility of the code using the datastore to identify which methods it should call. @kevinrizza, please comment if I misspoke.

I personally would prefer to keep the logic in the existing method for simplicity and refactor the datastore in a future JIRA.

Reply:
I don't understand what this has to do with a future refactoring. This is something you can do in this PR. If you want to have two APIs, please leave the old API as it is and introduce the new functionality in a new API.